### ISSUE TITLE
Character Controller 3D getting stuck/slowing down on trim meshes

### ISSUE THREAD

**Issue by GitUser442** [godotengine/godot#89784]
### Tested versions

- Reproduceable in 4.2.1

### System information

Godot v4.2.1.stable - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1650 SUPER (NVIDIA; 31.0.15.3770) - AMD Ryzen 5 3500 6-Core Processor (6 Threads)

### Issue description

When walking along a lightly protruding mesh (window/door trim in my case), the Character Body 3D will slow down or stop completely. I think something with collisions is to blame. I noticed the height will effect how bad the bug is. I am using the Capsule collision shape. The building model and its collision was imported from blender as a .blend file.

https://github.com/godotengine/godot/assets/164534337/4fd06239-b4f9-4de9-9cfd-fdf87f4f0dca

![Screenshot (221)](https://github.com/godotengine/godot/assets/164534337/925fe8be-e3d8-44b1-9d3e-151642e8ab5c)

### Steps to reproduce

Walk along trim type meshes.
I am using Jolt physics and a character controller for the AssetLib by expressobits.

### Minimal reproduction project (MRP)

https://www.mediafire.com/file/qbj951stjyh5o5e/Project4+-+Copy.zip/file

**Comment by Calinou** [comment]
>  I am using Jolt physics and a character controller for the AssetLib by expressobits.

Can you reproduce this issue with GodotPhysics? If not, please report this on https://github.com/godot-jolt/godot-jolt instead.

**Comment by GitUser442** [comment]
> This is the kind of thing [godotengine/godot-proposals#3603](https://github.com/godotengine/godot-proposals/issues/3603) could resolve.
> 
> > I am using Jolt physics and a character controller for the AssetLib by expressobits.
> 
> Can you reproduce this issue with GodotPhysics? If not, please report this on https://github.com/godot-jolt/godot-jolt instead.

Yes! Reproduceable on Default, Jolt, and Godot. It felt like Jolt was the least buggy.

**Comment by laviphon** [comment]
Can confirm this bug with with either Jolt or GodotPhysics.

Seems to happen when moving at certain angles where multiple meshes meet, messing around with the Save Margin value in CharacterBody3D can help alleviate it slightly but we're not able to completely mitigate it.
